### ISSUE TITLE
[DataGrid] Allow filtering on multiple values or include option to include accents in filtering

### ISSUE THREAD

**Issue by mikedidomizio** [mui-org/material-ui-x#2172]
I'd like the ability to allow filter on a row where cell contains a value that may have accented characters.  As a user, I might know the spelling of a name for example, but not know when/where to use the correct accents.

- [X] I have searched the [issues](https://github.com/mui-org/material-ui-x/issues) of this repository and believe that this is not a duplicate.

## Summary 💡

For example: a cell value is `Julio César Chávez`.  I should be able to allow the user to be able to search for that without any accents or some accents.

`Julio Cesar Chávez` and `Julio César Chavez` or `Julio Cesar Chavez` should also return that row.

My initial thoughts are it could be implemented in two ways:

1. Allow a custom filter method where I can transform the value how I like, and return if it matches or not

ex.

``` 
// would require more complexity if you have multiple operators
return (originalValue === searchCriteria || transformedValue === searchCriteria)
```

2. Add to the column definition
```
  searchIncludesAccented: true
```

I like the first way as it allows more flexibility.

## Examples 🌈

None other than what I've written above

## Motivation 🔦

Depending on the implementation, it could allow developers to have more advanced filtering on the value.

## Order id 💳

None

**Comment by m4theushw** [comment]
Are you using the "equals" or the "contains" operator? 

The example in 1. is an equality comparison but the built-in "equals" operator already allows to search without accents and match values with accents.

**Comment by mikedidomizio** [comment]
## TL;DR I have resolved this and I'm closing it

I tried a combination of both equals and contains for this.

I don't think it works the way I was hoping, but I did figure out that datagrid already supports what I was hoping for (custom operators), I was just not reading into the documentation well enough.

I wanted the user to be able to use a non-accented filter value and values with accents, would be compared and return matches.

Here is the filter working as expected with a name with an accent

![image](https://user-images.githubusercontent.com/5728044/126718644-169f5eda-8112-430f-b40d-f91d30897059.png)

Here is the filter working (I think as expected) where the filter has no accents and returns nothing.  I want to return the same result

![image](https://user-images.githubusercontent.com/5728044/126718679-3ac9dde9-ca29-47a9-a2f7-4e34285930d8.png)

**My fix**

So the functionality I was hoping datagrid had, does indeed have it.  I need a custom operator, which I did read but I wasn't picking up on it that it was what I wanted.  Here is my custom operator

```
  /**
   * Takes the user's filter value, changes it to lowercase, removes accents and compares it against the cell value lowercase
   * Useful for names
   */
  const containsWithoutAccentsAndLowerCase = [
    {
      label: 'contains',
      value: 'contains',
      getApplyFilterFn: (filterItem: GridFilterItem) => {
        if (
          !filterItem.columnField ||
          !filterItem.value ||
          !filterItem.operatorValue
        ) {
          return null;
        }

        return (params: { value: string }): boolean => {
          const valLowerCase = params.value.toLowerCase();
          // strip the filter value of accents
          const valLowerCaseNoAccents = valLowerCase
            .normalize('NFD')
            .replace(/[\u0300-\u036f]/g, '');
          const filterItemLowerCase = filterItem.value.toLowerCase();
          return (
            valLowerCase.includes(filterItemLowerCase) ||
            valLowerCaseNoAccents.includes(filterItemLowerCase)
          );
        };
      },
      InputComponent: MatchingMaterialUiInputField,
    },
  ];
```

and the outcome

![chrome_3pGfkcqj9x](https://user-images.githubusercontent.com/5728044/126719083-f4629231-9ccf-4dad-878b-ff01808907a9.png)
